homeassistant/components/homekit/__init__.py

```diff
@@ -124,6 +129,33 @@ def get_accessory(hass, state, aid, config):
     elif state.domain == 'lock':
         a_type = 'Lock'
 
+    elif state.domain == 'media_player':
+        features = state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+
+        supported_modes = []
+        if features & (SUPPORT_TURN_ON | SUPPORT_TURN_OFF):
+            supported_modes.append(ON_OFF)
+        if features & (SUPPORT_PLAY | SUPPORT_PAUSE):
+            supported_modes.append(PLAY_PAUSE)
+        if features & (SUPPORT_PLAY | SUPPORT_STOP):
+            supported_modes.append(PLAY_STOP)
+        if features & SUPPORT_VOLUME_MUTE:
+            supported_modes.append(TOGGLE_MUTE)
+
+        config_modes = config.get(CONF_MODE, DEFAULT_MEDIA_PLAYER_MODES)
+
+        validated_modes = []
+        for mode in config_modes:
+            if mode in supported_modes and mode not in validated_modes:
+                validated_modes.append(mode)
+            if mode not in supported_modes:
+                _LOGGER.warning('The entity "%s" does not support '
+                                'mode: "%s", supported modes are: %s',
+                                state.entity_id, mode, supported_modes)
+        if validated_modes is not None:
+            config[CONF_MODE] = validated_modes
```

Comment:
Good question. I haven't tested it, but you should be able to do these checks before the `super` call in the `type`. If the setup should be aborted, a return should do the trick.

Reply:
If we do before the `super` call, how to get `supported_features` since `self.hass` isn't initialized yet?

Reply:
You should be able to access all necessary attributes through `args`. However this would be pretty ugly.
What do think of only moving it to a separate function in `util` instead, so it doesn't clutter `__init__.py` and `get_accessory` that much?